src/frontend/js/modules/authentication.js

```diff
@@ -7,16 +7,39 @@ angular.module('esn.authentication', ['esn.http'])
     const SERVER_TTL = 60 * 1000;
     const NETWORK_LAG = 15 * 1000;
     const CACHE_MAX_TTL = SERVER_TTL - NETWORK_LAG;
+    const JWT_CACHE_MAX_TTL = 60 * 1000 * 15; // 15 minutes
```

Comment:
Why 15 minutes?

Reply:
@tuanlc normally the JWT never expires and in ESN we don't include the expiration date in the tokens. so I chose a 15 minutes duration. I'm open to ideas.
I checked [this](https://gist.github.com/soulmachine/b368ce7292ddd7f91c15accccc02b8df#:~:text=A%20JWT%20token%20that%20never,NOT%20be%20accepted%20for%20processing.) it could be helpful

Reply:
> normally the JWT never expires and in ESN we don't include the expiration date in the tokens.

This is a bad practice. Anyway, if the token is never expired, so why don't cache it forever but 15 minutes?

Reply:
A JWT that never expires is too dangerous. I think we should create an issue to resolve this problem first, and we should ask @MichaelBailly about it too.

And after that, what's ideal is we can get the exp date from the token and set the cache's TTL accordingly.

Reply:
Please decorrelate. In //,

1/ in calendar, support never-expiring JWT as well as expiring JWT
2/ in esn, add support for expiring JWT, and make sure we can renew the JWT with a JWT.

renew the JWT with a JWT: as a background job, when my jwt token is about to expire, I can call /api/jwt/generate, with the current JWT, to get a new JWT.